tests/virt_autotest/libvirt_extend_storage_lvm.pm

```diff
@@ -0,0 +1,171 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Virtualization Extend LVM Storage pool / volume test
+#
+# Test flow:
+# - Check a additional free hard disk
+# - Wipe hard disk clean
+# - Partition test disk
+# - Create a pv and display result
+# - Create a vg named 'lvm_vg' and display result
+# - Create a storage pool (lv) named 'guest_image_lvm'
+# - Create a logical volume (lv) size 1G
+# - Attach a logical volume (lv) to guest systems
+# - Clone a logical volume (lv)
+# - Cleanup
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "virt_feature_test_base";
+use virt_autotest::utils;
+use virt_autotest::common;
+use strict;
+use warnings;
+use testapi;
+use utils;
+use virt_utils;
+use version_utils 'is_sle';
+
+our $lvm_vg_name = 'lvm_vg';
+our $lvm_pool_name = 'guest_image_lvm';
+our $timeout = 180;
+sub run_test {
+    my ($self) = @_;
+
+    record_info "Prepare Guest Systems";
+    foreach (keys %virt_autotest::common::guests) {
+        start_guests() unless is_guest_online($_);
+    }
+
+    ## About LVM volumes management
+    my ($sys_disk, $virt_disk, $dev, $lvm_disk);
+    # Check with all existed Hard disks
+    my @disks = split(/\n/, script_output("lsblk -n -l -o NAME -d -e 7,11"));
+    my $scalar = @disks;
+    # Requires at least 2 physical hard disks for LVM Storage test
+    if (($scalar eq 1) || get_var('KEEP_DISKS')) {
+        record_info("WARNING", "Requires at least 2 physical hard disks for LVM Storage test\n", result => 'softfail');
+        return;
+    }
+    # Find out the used(mounted) hard disks
+    $sys_disk = script_output 'lsblk -rnoPKNAME $(findmnt -nrvoSOURCE /)';
+    record_info "System Disk:", "$sys_disk";
+    record_info("List Block Devices", script_output("lsblk"));
+    # Use a additional free hard disk for LVM volumes
+    $dev = "/dev/";
+    foreach my $disk (@disks) {
+        if ($disk ne $sys_disk) {
+            if (is_sle '15+') {
+                $virt_disk = script_output 'lsblk -rnoPKNAME $(findmnt -nrvoSOURCE /var/lib/libvirt)';
```

Comment:
I do not think 
```
findmnt -nrvoSOURCE /var/lib/libvirt
```
is good enough.
It should not depend on sle version, because if you install sle as a normal server without kvm/xen role, then there will be no separate mount point for libvirt. Although you can install kvm/xen role later, but there will be no sperate mount point for libvirt folder.
So better solution is to find the mount point of libvirt's parent folder until found, for example:
```
while (findmnt xxx fails) {
     xxx = xxx's parent folder
}
```
This is better and, at the same time, easy to implement.

Reply:
1st, we should focus on SLES with KVM and XEN system role for our virt_autotest without sle normal server. 
2nd, from SLE15+, we have only deployed  /var/lib/libvirt for libvirt staff on KVM|XEN system role without SLE12+. 
So, find out the $libvit_disk, should be depended on sle version.

Reply:
What you mean is about "test scenario". Finding mount point is generic operation which can be done better because there is no guarantee that a folder has its own mount point. Current code sacrifices portablity and compatibility.

Reply:
yes, good point. make sense now. check with $sys_disk and $libvirt_disk to find out free or unused disks, this solution  sacrifices portability and compatibility.

Reply:
update code to find out unused hard disk